os/board/rtl8730e/src/component/wifi/inic/inic_ipc_host_api.c

```diff
@@ -323,13 +333,37 @@ static void inic_ipc_print_int_hdl(VOID *Data, u32 IrqStatus, u32 ChanNum)
 	(void) IrqStatus;
 	(void) ChanNum;
 
+	static km4log_msg_t message_event = { 0 };
+
+	/* receive a log message over IPC */
 	PIPC_MSG_STRUCT ipc_recv_msg = (PIPC_MSG_STRUCT)ipc_get_message(IPC_NP_TO_AP, IPC_N2A_NP_LOG_CHN);
 	char *tmp_buffer = (char *)ipc_recv_msg->msg;
 	DCache_Invalidate((u32)tmp_buffer, ipc_recv_msg->msg_len);
-	
-	/* Print out buffer */
-	lldbg_noarg("%s",tmp_buffer);
 
+	/* pick a buffer to fill. for rapid processing, assume that the higher priority task has already consumed the buffer, otherwise overwrite it */
+	u8 buf_idx = g_inic_ipc_logging_buf_ctr;
+
+	/* copy the message */
+	strcpy((char*)g_inic_ipc_logging_buf[buf_idx], tmp_buffer);
+
+	/* fill message struct */
+	message_event.buffer = (void*)g_inic_ipc_logging_buf[buf_idx];
+	message_event.buffer_len = NP_MAX_BUFFER_SIZE;
+
+	/* use mq_send via osif api directly in ISR instead of semaphore-based */
+	if(g_km4_log_queue == NULL || false == osif_msg_send(g_km4_log_queue, &message_event, 0)) {
+		/* mixlog queue handle was invalid, or sending to queue failed, clear the memory here. */
+		DBG_8195A("queue hndl is null or send failed\n");
+
+		/* set the first byte to null to mark this buffer as usable */
+		g_inic_ipc_logging_buf[buf_idx][0] = 0;
```

Comment:
How about changing the rsvd in the struct member to some meaningful flag to indicate the status of the buffer, instead of setting first byte to 0?
Besides, there is no condition checking this statement: "buffer marked as usable". If then, what is the reason of doing so?